src/DI/ContainerBuilder.php

```diff
@@ -411,6 +417,17 @@ private function resolveImplement(ServiceDefinition $def, $name)
 	}
 
 
+	private function getTypeHintForParameter(\ReflectionParameter $param)
+	{
+		if (PHP_VERSION_ID >= 70000 && $param->hasType()) {
+			$type = $param->getType();
+			return $type->isBuiltin() ? (string) $type : '\\' . $type;
+		} else {
+			return $param->isArray() ? 'array' : PhpReflection::getPropertyType($param);
```

Comment:
Would it be a problem if the callable support would get merged after/in this commit after all? I can do it (actually have it already, including test), but I built it on this PR - so it would be easier to get it accepted after (so I don't have to rebase this PR).

Or we can add it in this PR and rename it to "add support for callable types and PHP 7 type hints" instead of just "add support for PHP7"?

But if you insist on doing first the callable and this, I can create another PR and then rebase this one - it's up to you, I'm not familiar with netter PR process that much. :)